### PR TITLE
Update the content of the try-it.md

### PR DESCRIPTION
**Pull request by Rozzii**
Update the content of the try-it.md to be aligned with the current state of the metal3 project.

### CODE REVIEW

File: try-it.md [hunk 1]
```diff
@@ -118,7 +118,6 @@ hosts. You can see the VMs using `virsh`.
 $ sudo virsh list
  Id    Name                           State
 ----------------------------------------------------
- 6     minikube                       running
```
Comment (fmuyassarov) [inline try-it.md:121]: Why is this removed? We still use minikube for CentOS based Metal3

File: try-it.md [hunk 2]
```diff
@@ -131,7 +130,7 @@ definition file used to create these host objects is in `bmhosts_crs.yaml`.
 $ kubectl get baremetalhosts -n metal3 -o wide
 NAME     STATUS   STATE   CONSUMER   BMC                         HARDWARE_PROFILE   ONLINE   ERROR
 node-0   OK       ready              ipmi://192.168.111.1:6230   unknown            true
-node-1   OK       ready              ipmi://192.168.111.1:6231   unknown            true
```
Comment (fmuyassarov) [inline try-it.md:134]: I think this part was showing unprovisioned (i.e. `Ready` BareMetalHosts), maybe that should be kept as it is?

File: try-it.md [hunk 3]
```diff
@@ -255,12 +329,26 @@ $ ./scripts/provision/controlplane.sh
 $ ./scripts/provision/worker.sh
 ```
 
+
 At this point, the `Machine` actuator will respond and try to claim a
-`BareMetalHost` for this `Machine`. You can check the logs of the actuator
-here:
+`BareMetalHost` for this `Machine`. You can check the logs of the actuator.
```
Comment (fmuyassarov) [inline try-it.md:334]:
```suggestion
`BareMetalHost` for this `Metal3Machine`. You can check the logs of the actuator.
```

File: try-it.md [hunk 4]
```diff
@@ -272,30 +360,73 @@ $ kubectl logs -n capm3-system pod/capm3-controller-manager-646878769b-qmrrp -c
...
 ```
 
-If you look at the yaml representation of the `Machine` object, you will see a
+Keep in mind that the suffix hashes e.g. `5d968ffd9d-8f6jz` are automatically generated and change in case of a different
+deployment.
+
+If you look at the yaml representation of the `Metal3Machine` object, you will see a
 new annotation that identifies which `BareMetalHost` was chosen to satisfy this
-`Machine` request.
+`Metal3Machine` request.
```
Comment (fmuyassarov) [inline try-it.md:368]: 👍

File: try-it.md [hunk 5]
```diff
@@ -272,30 +360,73 @@ $ kubectl logs -n capm3-system pod/capm3-controller-manager-646878769b-qmrrp -c
...
+
+First list the machine objects present in the `metal3` namespace:
```
Comment (fmuyassarov) [inline try-it.md:370]:
```suggestion
First list the metal3Machine objects present in the `metal3` namespace:
```

File: try-it.md [hunk 6]
```diff
@@ -272,30 +360,73 @@ $ kubectl logs -n capm3-system pod/capm3-controller-manager-646878769b-qmrrp -c
...
 ```sh
-$ kubectl get machine centos -n metal3 -o yaml
+$ kubectl get metal3machines -n metal3
+NAME                       PROVIDERID                                      READY   CLUSTER   PHASE
+test1-controlplane-ssd56   metal3://d4848820-55fd-410a-b902-5b2122dd206c   true    test1
+test1-workers-gjcts        metal3://ee337588-be96-4d5b-95b9-b7375969debd   true    test1
+```
 
+Based on the name of the `Metal3Machine` objects you can check the yaml representation of the object and
+see which `BareMetalHost` was chosen.
```
Comment (fmuyassarov) [inline try-it.md:380]:
```suggestion
see from its annotation which `BareMetalHost` was chosen.
```

File: try-it.md [hunk 7]
```diff
@@ -325,31 +456,45 @@ $ kubectl scale machinedeployment test1-md-0 --replicas=0
 > warning "Warning"
 > control-plane and cluster are very tied together. This means that you are not able to deprovision the control-plane of a cluster and then provision a new one within the same cluster. Therefore, in case you want to deprovision the control-plane you need to **deprovision the cluster** as well and provision both again.
 
-Below, it is shown how the deprovisioning can be executed in a more manual way by just deleting the proper Custom Resources (CR)
+Below, it is shown how the deprovisioning can be executed in a more manual way by just deleting the proper Custom Resources (CR).
+
+The order of deletion is:
+1. Metal3Machine objects of the workers
+2. Machine objects of the workers
```
Comment (fmuyassarov) [inline try-it.md:463]: I think it is vice versa

File: try-it.md [hunk 8]
```diff
@@ -325,31 +456,45 @@ $ kubectl scale machinedeployment test1-md-0 --replicas=0
...
+3. Metal3Machine objects of the control plane
+4. Machine objects of the control plane
```
Comment (fmuyassarov) [inline try-it.md:465]: Same here

File: try-it.md (hunk 7 above)
Comment (Rozzii) [reply try-it.md:463]: Yes it was typo.

File: try-it.md (hunk 8 above)
Comment (Rozzii) [reply try-it.md:465]: Yes it was typo.

File: try-it.md (hunk 2 above)
Comment (Rozzii) [reply try-it.md:134]: I have accidentally copied the the wrong output, actually before provisioning it looks like this:
```sh
$ kubectl get baremetalhosts -n metal3 -o wide
NAME     STATUS   STATE   CONSUMER   BMC                         HARDWARE_PROFILE   ONLINE   ERROR
node-0   OK       ready              ipmi://192.168.111.1:6230   unknown            true
node-1   OK       ready              redfish+http://192.168.111.1:8000/redfish/v1/Systems/a1cd44ba-c6db-49ac-bb07-56d4fbc5380f   unknown            true
```

File: try-it.md (hunk 1 above)
Comment (Rozzii) [reply try-it.md:121]: I have installed the dev-env with CentOS images but this minikube vm still didn't come up, from the source it looks to me that it only comes up if I explicitly set the EPHEMERAL_CLUSTER environment variable to have the value "minikube". I will document both case when EPHEMERAL_CLUSTER is set to "minikube" and when it is set to "kind".

Comment (fmuyassarov) [reply try-it.md:121]: Did you have CentOS OS on the host VM where metal3-dev-env was running or was CentOS used for your target VMs?

Comment (Rozzii) [reply try-it.md:121]: Host VM was Ubuntu 20 and I used CentOS for target VMs.

Comment (fmuyassarov) [reply try-it.md:121]: ok that explains :) So there two different OS that we test metal3-dev-env on
1. Host VM/Server on CentOS, while target can be Ubuntu or CentOS, etc.
2. Host VM/Server on Ubuntu, while target can be Ubuntu or CentOS, etc.

The way we run k8s clusters for above two scenarios is different. For CentOS we use minikube cluster, for Ubuntu Kind cluster. As such, when your host is CentOS, you should be able to see three libvirt VMs. Since you are using Ubuntu, the source k8s cluster was created via Kind, which is why you don't see minikube VM. To control what tool should be used for creating source k8s cluster, we use EPHEMRAL_CLUSTER variable, which is minikube by default for CentOS and kind for Ubuntu.

Comment (Rozzii) [reply try-it.md:121]: Thanks for the clarification, I have incorporated your explanation to the try-it.md with a new commit, I think it could be useful for beginners.

File: try-it.md [hunk 9]
```diff
@@ -112,15 +115,44 @@ $ vim config_$(whoami).sh
...
-hosts. You can see the VMs using `virsh`.
+hosts.
+
+There are two different host OSs that metal3-dev-env setup process is tested on.
+
+  1. Host VM/Server on CentOS, while target can be Ubuntu or CentOS, Cirros, FCOS.
+  2. Host VM/Server on Ubuntu, while target can be Ubuntu or CentOS, Cirros, FCOS.
+
+The way  k8s clusters is running in the above two scenarios is different. For CentOS `minikube` cluster is used as the source cluster, for Ubuntu a `kind` cluster is being cerated.
+As such, when the host (where the `make` command was issued) is CentOS, there should three libvirt VMs and one of them should be a `minikube` VM.
```
Comment (fmuyassarov) [inline try-it.md:126]:
```suggestion
As such, when the host (where the `make` command was issued) is CentOS, there should be three libvirt VMs and one of them should be a `minikube` VM.
```

File: try-it.md [hunk 10]
```diff
@@ -145,97 +177,172 @@ kind: BareMetalHost
...
+      - tsc_known_freq
+      - umip
+      - vme
+      - vmx
+      - vnmi
+      - vpid
+      - x2apic
+      - xsave
+      - xsaveopt
+      - xtopology
```
Comment (fmuyassarov) [inline try-it.md:276]: Maybe we don't need to include all the flags here and instead just mention that this is concatenated list. Because, this is try-it or sort of quick-start guide, which is meant to be short and should be easy to work through. I think readers won't really read these flags when they try to setup metal3-dev-env.

**Review by fmuyassarov — COMMENTED** [review]
Looks good, some more comments inline

File: try-it.md (hunk 10 above)
Comment (Rozzii) [reply try-it.md]: I agree.

File: try-it.md [hunk 11]
```diff
@@ -272,30 +330,73 @@ $ kubectl logs -n capm3-system pod/capm3-controller-manager-646878769b-qmrrp -c
...
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+    kind: Metal3Machine
+    name: test1-workers-gjcts
+    namespace: metal3
+    uid: 9adaec5f-f72b-4674-9f8f-1dc6c9039755
+...
 ```
 
+Based on the result of  the query `test1-75678f6485-z928j` ClusterAPI `Machine` object is backed by
```
Comment (fmuyassarov) [inline try-it.md:397]:
```suggestion
Based on the result of the query `test1-75678f6485-z928j` ClusterAPI `Machine` object is backed by
```

File: try-it.md [hunk 12]
```diff
@@ -325,31 +426,45 @@ $ kubectl scale machinedeployment test1-md-0 --replicas=0
...
+object thus in order to avoid reprovisioning of the `Machine` object the  `MachineDeployment` has to be deleted instead of the `Machine` object in the case of `test1-workers-gjcts`.
 
 ```sh
-$ kubectl delete machine test1-md-0-m87bq -n metal3
-machine.cluster.x-k8s.io "test1-md-0-m87bq" deleted
+$ kubectl delete machinedeployment test1 -n metal3
+machinedeployment.cluster.x-k8s.io "test1" deleted
+
+$ kubectl delete metal3machine test1-workers-gjcts -n metal3
+machine.cluster.x-k8s.io "test1-75678f6485-z928j" deleted
```
Comment (fmuyassarov) [inline try-it.md:446]: When MachineDeployment is deleted, Machine object should also get deleted I assume. As such, probably this step is redundant?

File: try-it.md [hunk 13]
```diff
@@ -325,31 +426,45 @@ $ kubectl scale machinedeployment test1-md-0 --replicas=0
...
+
+$ kubectl delete machine test1-m77bn -n metal3
+machine.cluster.x-k8s.io "test1-m77bn" deleted
 
-$ kubectl delete machine test1-controlplane-0 -n metal3
-machine.cluster.x-k8s.io "test1-controlplane-0" deleted
+$ kubectl delete metal3machine test1-controlplane-ssd56 -n metal3
+machine.cluster.x-k8s.io "test1-controlplane-ssd56" deleted
```
Comment (fmuyassarov) [inline try-it.md:452]: No need for including Metal3Machine deletion step, because when Machine is deleted, corresponding M3Machine will get deleted automatically. That's a reason why in the original document we were showing only deletion of Machine objects, which normally causes deletion of M3Machine, and BMH object deprovisioning.

**Review by fmuyassarov — COMMENTED** [review]
Final comments :)

**Comment by fmuyassarov** [comment]
Not sure what's the issue with the Prow that it doesn't automatically merge PRs on this repo. Not to block others meanwhile, I'm going to merge this myself.
